src/dashboard/src/main/management/commands/rebuild_aip_index_from_storage_service.py

```diff
@@ -114,109 +117,72 @@ def handle(self, *args, **options):
-                    )
-                )
+        aips_to_index_count = len(aips_to_index)
 
-            self.info("Cleaning up")
-            shutil.rmtree(temp_dir)
-            self.info("Indexing complete.")
-            return
+        # If there's nothing to index, log error and quit.
+        if not aips_to_index_count:
+            self.error("No AIPs found to index. Quitting.")
+            sys.exit(1)
 
+        # Setup es_client and delete indices if required.
+        es_client = setup_es_for_aip_reindexing(self, options["delete_all"])
         self.info("Rebuilding 'aips' and 'aipfiles' indices")
 
+        # Index packages.
         packages_not_indexed = []
-
-        # Index non-deleted AIPs associated with this pipeline.
-        aips = storageService.get_file_info(package_type="AIP")
-        filtered_aips = am.filter_packages_by_status_and_pipeline(
-            aips, pipeline_uuid=pipeline_uuid
-        )
-        filtered_aip_count = len(filtered_aips)
-
         aip_indexed_count = 0
-        for aip in filtered_aips:
+        for aip in aips_to_index:
+            is_aic = False
+            if aip["package_type"] == "AIC":
+                is_aic = True
             index_success = self.process_package(
-                es_client, aip, temp_dir, delete_before_reindexing
+                es_client, aip, temp_dir, delete_before_reindexing, is_aic=is_aic
             )
-            if index_success is True:
+            if index_success:
                 aip_indexed_count += 1
             else:
                 packages_not_indexed.append(aip["uuid"])
 
-        # Index non-deleted AICs associated with this pipeline.
-        aics = storageService.get_file_info(package_type="AIC")
-        filtered_aics = am.filter_packages_by_status_and_pipeline(
-            aics, pipeline_uuid=pipeline_uuid
-        )
-        filtered_aic_count = len(filtered_aics)
-
-        aic_indexed_count = 0
-        for aic in filtered_aics:
-            index_success = self.process_package(
-                es_client, aic, temp_dir, delete_before_reindexing, is_aic=True
-            )
-            if index_success is True:
-                aic_indexed_count += 1
-            else:
-                packages_not_indexed.append(aic["uuid"])
-
         # Clean up and report on packages indexed.
         self.info("Cleaning up")
         shutil.rmtree(temp_dir)
 
         if packages_not_indexed:
-            indexing_stats = "{0} of {1} AIPs and {2} of {3} AICs".format(
-                aip_indexed_count,
-                filtered_aip_count,
-                aic_indexed_count,
-                filtered_aic_count,
-            )
-            self.info(
-                "Indexing complete. Indexed {0}. Packages not indexed: {1}.".format(
-                    indexing_stats, ", ".join(packages_not_indexed)
+            self.error(
+                "Indexing complete. Indexed {0} of {1} AIPs/AICs. Packages not indexed: {1}.".format(
```

Comment:
```suggestion
                "Indexing complete. Indexed {} of {} AIPs/AICs. Packages not indexed: {}.".format(
```
I think you want `{2}` here! :wink: that being said it might be a good (inadvertent) demonstration of where we can let the positional arguments do the work, and avoid explicit indexes where possible.

I'd chose `{}` in this case. For other instances if you're reusing variables during string concatenation, then the named approach works quite nicely as a self-describing and very explicit option:
```python
            self.error(
                "Indexing complete. Indexed {demo_1} of {demo_2} AIPs/AICs. Packages not indexed: {demo_3}.".format(
                    demo_1=aip_indexed_count,
                    demo_2=aips_to_index_count,
                    demo_3=", ".join(packages_not_indexed),
                )
            )
```
There's a whole bunch of string concatenation methods but I tend to stick with this one for AM and personally.

Reply:
Good catch! This named option is nice. It reminds me of f-strings, which I tend to prefer when possible.